wicket/src/ui/panes/update.rs

```diff
@@ -0,0 +1,293 @@
+        UpdatePane {
+            tree_state,
+            items: ALL_COMPONENT_IDS
+                .iter()
+                .map(|id| TreeItem::new(*id, vec![]))
+                .collect(),
+            help: vec![
+                ("OPEN", "<RIGHT>"),
+                ("CLOSE", "<LEFT>"),
+                ("SELECT", "<UP/DOWN>"),
+            ],
+            rect: Rect::default(),
+            title_rect: Rect::default(),
+            table_headers_rect: Rect::default(),
+            contents_rect: Rect::default(),
+            help_rect: Rect::default(),
+            popup_open: false,
+        }
+    }
+
+    pub fn draw_update_missing_popup(
+        &mut self,
+        state: &State,
+        frame: &mut Frame<'_>,
+    ) {
+        let popup = Popup {
+            header: Text::from(vec![Spans::from(vec![Span::styled(
+                " MISSING UPDATE BUNDLE",
+                style::header(true),
+            )])]),
+            body: Text::from(vec![
+                Spans::from(vec![Span::styled(
+                    " Use the following command to transfer an update: ",
+                    style::plain_text(),
+                )]),
+                "".into(),
+                Spans::from(vec![
+                    Span::styled(" cat", style::plain_text()),
+                    Span::styled(" $UPDATE", style::popup_highlight()),
+                    Span::styled(".zip | ssh", style::plain_text()),
+                    Span::styled(" $IPV6_ADDRESS", style::popup_highlight()),
+                    Span::styled(" upload", style::plain_text()),
+                ]),
+            ]),
+            buttons: vec![ButtonText { instruction: "CLOSE", key: "ESC" }],
+        };
+        let full_screen = Rect {
+            width: state.screen_width,
+            height: state.screen_height,
+            x: 0,
+            y: 0,
+        };
+        frame.render_widget(popup, full_screen);
+    }
+
+    fn update_items(&mut self, state: &State) {
+        let versions = state.update_state.artifact_versions.clone();
+
+        self.items = state
+            .update_state
+            .items
+            .iter()
+            .map(|(id, states)| {
+                let children: Vec<_> = states
+                    .iter()
+                    .map(|(artifact, s)| {
+                        let version = artifact_version(artifact, &versions);
+                        let spans = vec![
+                            Span::styled(
+                                artifact_title(*artifact),
+                                style::selected(),
+                            ),
+                            Span::styled("UNKNOWN", style::selected_line()),
+                            Span::styled(version, style::selected()),
+                            Span::styled(s.to_string(), s.style()),
+                        ];
+                        TreeItem::new_leaf(align_by(
+                            0,
+                            25,
```

Comment:
What does the 25 mean here? Could you add a comment?

There's another 25 below so maybe this should be defined as a constant.

Reply:
Good catch. It's the max number of columns per `Span`. I'll add a constant.

Reply:
fixed.